packages/sd-jwt-vc/src/test/index.spec.ts

```diff
@@ -53,3 +76,108 @@ describe('App', () => {
     expect(encodedSdjwt).rejects.toThrowError();
   });
 });
+
+describe('Revocation', () => {
+  const { signer, verifier } = createSignerVerifier();
+  const sdjwt = new SDJwtVcInstance({
+    signer,
+    signAlg: 'EdDSA',
+    verifier,
+    hasher: digest,
+    hashAlg: 'SHA-256',
+    saltGenerator: generateSalt,
+    statusListFetcher(uri: string) {
+      // we emulate fetching the status list from the uri. Validation of the JWT is not done here in the test but should be done in the implementation.
+      return Promise.resolve(statusListJWT);
+    },
+    statusValidator(status: number) {
+      // we are only accepting status 0
+      if (status === 0) return Promise.resolve();
+      throw new Error('Status is not valid');
+    },
+  });
+
+  test('Test with a non revcoked credential', async () => {
+    const claims = {
+      firstname: 'John',
+      status: {
+        status_list: {
+          uri: 'https://example.com/status-list',
+          idx: 0,
+        },
+      },
+    };
+    const expectedPayload: SdJwtVcPayload = { iat, iss, vct, ...claims };
+    const encodedSdjwt = await sdjwt.issue(expectedPayload);
+    const result = await sdjwt.verify(encodedSdjwt);
+    expect(result).toBeDefined();
+  });
+
+  test('Test with a revoked credential', async () => {
+    const claims = {
+      firstname: 'John',
+      status: {
+        status_list: {
+          uri: 'https://example.com/status-list',
+          idx: 1,
+        },
+      },
+    };
+    const expectedPayload: SdJwtVcPayload = { iat, iss, vct, ...claims };
+    const encodedSdjwt = await sdjwt.issue(expectedPayload);
+    const result = sdjwt.verify(encodedSdjwt);
+    expect(result).rejects.toThrowError('Status is not valid');
```

Comment:
As also mentioned in the Credo PR to @lukasjhan, it would be great if the verify method could return an object of validations (as we had in our sd-jwt implementation before merging them together).

For us it's relevant to know which validations succeed and which don't (sometimes it's ok that a cred is revoked). Sometimes it's ok that it's expired. It would be nice to not throw an error but return an validation object

Reply:
@cre8 @berendsliedrecht 
What do you guys think? 

If we're going to change it, I think it would be better to go through a breaking change before v1.0.

Reply:
I agree with @TimoGlastra. We have had this pattern provided by some previous libraries as well and it works quite nice. As Timo mentioned, some validation failures are okay.

Reply:
Opened #228 for this